src/server/pfs/cmds/cmds.go

```diff
@@ -1199,6 +1201,30 @@ $ {{alias}} 'foo@master:/test\[\].txt'`,
 			}
 			defer c.Close()
 			defer progress.Wait()
+			// TODO: Decide what progress should look like in the recursive case. The files are downloaded in a batch in 2.x.
+			if recursive {
+				if outputPath == "" {
+					return errors.Errorf("an output path needs to be specified when using the --recursive flag")
+				}
+				// Check that the path matches one directory / file.
+				_, err := c.InspectFile(file.Commit, file.Path)
+				if err != nil {
+					return err
+				}
+				r, err := c.GetFileTAR(file.Commit, file.Path)
+				if err != nil {
+					return err
+				}
+				dir, err := ioutil.TempDir("", "pachyderm_get_file_recursive")
+				if err != nil {
+					return err
+				}
+				defer os.RemoveAll(dir)
+				if err := tarutil.Import(dir, r); err != nil {
+					return err
+				}
+				return os.Rename(path.Join(dir, file.Path), outputPath)
```

Comment:
Do we really just clobber everything in the entire directory? Overwriting the files we are exporting is expected, but a hard replace of the directory node itself is going to surprise someone and cause data loss.

Reply:
Okay, I was thinking in terms of syncing up the PFS state with the local state, but that seems like the incorrect way to think about this functionality (considering how recursive put file works). I'll change this to just create / overwrite the files that are returned.

Reply:
Cool, everything else looks good. I will approve.  FWIW this is definitely more useful for certain use cases; I just don't think it can be a good default.